### PR TITLE
PICARD-2582: Get rid of legacy distutils imports in setup.py

### PR DESCRIPTION
**Pull request by phw**
# Summary

* This is a…
  * [x] Bug fix
  * [ ] Feature addition
  * [x] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): PICARD-2582

Picard's `setup.py` was still importing some legacy distutils components to achieve some functionality.

Also these distutils imports are imported before setuptools gets imported, which is actively discouraged as setuptools might provide modified distutils.

# Solution

Replace all distutils imports with proper replacements (see also [PEP 632](https://peps.python.org/pep-0632/))

# Action

This needs a newer setuptools which provides the `build` command implementation (was added sometime in August this year).

### CODE REVIEW

**Comment by phw** [comment]
The `command.build` was added to setupttools 62.4.0, which was released in June 13th '22, so that's the minimum supported version with this patch.

File: setup.py [hunk 1]
```diff
@@ -39,14 +39,12 @@
 
 
 import datetime
-from distutils import log
-from distutils.command.build import build
-from distutils.dep_util import newer
-from distutils.spawn import find_executable
 import glob
 from io import StringIO
+import logging as log
```
Comment (zas) [inline setup.py:44]: Is this import needed?

Comment (phw) [reply setup.py:44]: Yes, log is used at various places in setup.py in the different commands. `distutils` provided its own `log` module, the recommended replacement is using the `logging` module instead, see https://peps.python.org/pep-0632/#migration-advice .

**Review by zas — APPROVED** [review]
LGTM

**Pull request merged**